src/recorder.cc

```diff
@@ -61,8 +69,13 @@ void Recorder::set_dump_count(unsigned int count) {
 Datum* Recorder::NewDatum(std::string title) {
   Datum* d = data_[index_];
   d->title_ = title;
-  d->vals_.resize(1);
-  d->shapes_.resize(1);
+  if (inject_sim_id_) {
```

Comment:
I don't think that we should be dogmatic about these things. No matter what way you cut it, this is minor. Also it is strictly in the implementation so anyone is welcome to change it in the future.